apps/site/assets/ts/helpers/service.ts

```diff
@@ -1,155 +1,197 @@
-  const endDateObject = new Date(endDate);
-  if ("service_date" in service) {
-    // Get unix timestamps
-    const serviceDateTime = new Date(service.service_date).getTime();
-    const startDateTime = startDateObject.getTime();
-    const endDateTime = endDateObject.getTime();
+    // if there's no rating end date, this is probably a CR or Ferry.
+    // don't use the rating in that case.
+    if (hasNoRating(service)) {
+      if (isInCurrentService(service, currentDate)) {
+        return "Current Schedules";
+      }
+      if (isInFutureService(service, currentDate)) {
+        return "Future Schedules";
+      }
+    } else {
+      const ratingStartDate = new Date(service.rating_start_date!);
+      const ratingEndDate = new Date(service.rating_end_date!);
+
+      if (
+        isInCurrentRating(service, currentDate) &&
+        service.typicality === "typical_service"
+      ) {
+        return `Current Schedules (${
+          service.rating_description
+        }, ends ${shortDate(ratingEndDate)})`;
+      }
 
-    if (serviceDateTime >= startDateTime && serviceDateTime <= endDateTime) {
-      return `ends ${formattedDate(endDateObject)}`;
+      if (isInFutureRating(service, currentDate)) {
+        return `Future Schedules (${
+          service.rating_description
+        }, starts ${shortDate(ratingStartDate)})`;
+      }
     }
 
-    if (serviceDateTime < startDateTime) {
-      return `starts ${formattedDate(startDateObject)}`;
-    }
-  }
-  return startToEnd(startDateObject, endDateObject);
+    return "Other Schedules";
+  });
+
+// sort the option groups
+const optGroupOrder: { [key: string]: number } = {
+  Current: 4,
+  Holiday: 3,
+  Future: 2,
+  Other: 1
 };
 
-type MonthInteger = number | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
+export const optGroupComparator = (groupA: string, groupB: string): number => {
+  const a = groupA.split(" ")[0];
+  const b = groupB.split(" ")[0];
+  return optGroupOrder[b] - optGroupOrder[a];
+};
 
-export const hasMultipleWeekdaySchedules = (
-  services: ServiceWithServiceDate[]
-): boolean =>
+export const hasMultipleWeekdaySchedules = (services: Service[]): boolean =>
   services.filter(
     service =>
       service.type === "weekday" && service.typicality !== "holiday_service"
   ).length > 1;
+
+const dayIntegerToString = (day: DayInteger): string =>
+  [
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday"
+  ][day - 1];
+
+const daysAreConsecutive = (
+  days: DayInteger[],
+  areConsecutive: boolean
+): boolean => {
+  if (!areConsecutive) return areConsecutive;
+  const [day, nextDay, ...rest] = days;
+  if (rest.length === 0) return day + 1 === nextDay;
+  return daysAreConsecutive([nextDay, ...rest], day + 1 === nextDay);
+};
+
+export const serviceDays = ({
+  type,
+  valid_days: validDays
+}: Service): string => {
+  if (type === "saturday" || type === "sunday") return "";
```

Comment:
What's the purpose of this check? It seems like it would be fine to get the service days of a saturday or sunday service, if we ever wanted to (it doesn't appear we ever go down this branch in practice, since the only calls to this function are guarded by `isMultipleWeekday`).

Reply:
You're right about not going down this branch in practice -- this is only used to describe valid days of service during weekdays -- e.g. `Tuesday - Friday` or `Monday, Friday` or other such patterns.